plugin/src/main/java/com/denizenscript/denizen/events/world/RaidStopsScriptEvent.java

```diff
@@ -0,0 +1,48 @@
+package com.denizenscript.denizen.events.world;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.utilities.world.RaidData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.raid.RaidStopEvent;
+
+public class RaidStopsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    public RaidStopsScriptEvent() {
+        registerCouldMatcher("raid stops");
+        registerSwitches("reason");
+    }
+
+    public RaidStopEvent event;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (path.switches.containsKey("reason") && !path.checkSwitch("reason", CoreUtilities.toLowerCase(event.getReason().name()))) {
```

Comment:
use `runGenericSwitchCheck`, remove the containsKey check, remove the toLower - it automates all of this stuff